### PR TITLE
web: add formatting to Timestamp component

### PR DESCRIPTION
**Pull request by sashaostrikov**
Follow up to https://github.com/sourcegraph/sourcegraph/pull/43429

We have a Timestamp component which should be used instead of parsing and formatting the date if possible.

## Test plan
New jest test added. Ran sg locally to check that the dates didn't change.

## App preview:

- [Web](https://sg-web-ao-ui-add-format-to-timestamp.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by mrnugget — COMMENTED** [review]
LGTM but I'd ask @sourcegraph/frontend-platform here

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.10 kb) | 0.00% (+0.25 kb) | 0.00% (+0.15 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/ui/add-format-to-timestamp/compare-report.html;tab=live_object) for a full comparison between the commits fe74238 and ca14f5d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by sashaostrikov** [comment]
> LGTM but I'd ask @sourcegraph/frontend-platform here

Actually, this change came from frontend platform, @valerybugakov gave me a hint that we have this component and we should use it instead of parsing the dates. And it was used previously, but I changed it previous PR.

File: client/web/src/components/time/Timestamp.tsx [hunk 1]
```diff
@@ -50,8 +58,8 @@ export const Timestamp: React.FunctionComponent<React.PropsWithChildren<Props>>
     const tooltip = useMemo(() => {
         const parsedDate = typeof date === 'string' ? parseISO(date) : new Date(date)
         const dateHasTime = date.toString().includes('T')
-        return format(parsedDate, `yyyy-MM-dd${dateHasTime ? ' pp' : ''}`)
-    }, [date])
+        return format(parsedDate, timestampFormat ?? `yyyy-MM-dd${dateHasTime ? ' pp' : ''}`)
```
Comment (valerybugakov) [inline client/web/src/components/time/Timestamp.tsx:61]: We can slightly improve here by moving the default format to the new `enum` to keep format strings in one place.

```tsx
export enum TimestampFormat {
    FULL_TIME = 'HH:mm:ss'
    FULL_DATE = 'yyyy-MM-dd'
    FULL_DATE_TIME = 'yyyy-MM-dd pp'
}

const defaultFormat = dateHasTime ? TimestampFormat.FULL_DATE_TIME : TimestampFormat.FULL_DATE
return format(parsedDate, timestampFormat ?? defaultFormat)

File: client/web/src/components/externalServices/ExternalServicePage.tsx [hunk 2]
```diff
@@ -387,11 +387,31 @@ const ExternalServiceSyncJobNode: React.FunctionComponent<ExternalServiceSyncJob
...
                 <div className="flex-shrink-1 flex-grow-0 mr-1">
                     {node.finishedAt === null && 'Not finished yet.'}
-                    {node.finishedAt !== null && <>Finished at {format(Date.parse(node.finishedAt), 'HH:mm:ss')}.</>}
+                    {node.finishedAt !== null && (
+                        <>
+                            Finished at{' '}
+                            <Timestamp
+                                date={node.finishedAt}
+                                preferAbsolute={true}
+                                timestampFormat={TimestampFormat.FULL_TIME}
```
Comment (valerybugakov) [inline client/web/src/components/externalServices/ExternalServicePage.tsx:410]: 🔥

**Review by valerybugakov — APPROVED** [review]

File: client/web/src/components/time/Timestamp.tsx (hunk 1 above)
Comment (sashaostrikov) [reply client/web/src/components/time/Timestamp.tsx:61]: thanks! this looks way better 👍🏻

**Pull request merged**